### PR TITLE
WT-6267 Fix not deleting prepared update when hs cursor is placed on another key

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: src/txn/txn.c [hunk 1]
```diff
@@ -925,9 +929,10 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
           true);
 
         if (ret == 0)
-            WT_ERR(__txn_append_hs_record(session, hs_cursor, &op->u.op_row.key, cbt->ref->page,
-              upd, commit, &fix_upd, &upd_appended));
-        else if (ret == WT_NOTFOUND && !commit) {
+            WT_ERR_NOTFOUND_OK(__txn_append_hs_record(session, hs_cursor, &op->u.op_row.key,
```
Comment (kommiharibabu) [inline src/txn/txn.c:932]: Better to add a comment to explain when we expect a NOTFOUND.

File: src/txn/txn.c [hunk 2]
```diff
@@ -925,9 +929,11 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
...
 
         if (ret == 0)
-            WT_ERR(__txn_append_hs_record(session, hs_cursor, &op->u.op_row.key, cbt->ref->page,
-              upd, commit, &fix_upd, &upd_appended));
-        else if (ret == WT_NOTFOUND && !commit) {
+            /* Not found if we cross the tree or key boundary. */
+            WT_ERR_NOTFOUND_OK(__txn_append_hs_record(session, hs_cursor, &op->u.op_row.key,
+                                 cbt->ref->page, upd, commit, &fix_upd, &upd_appended),
+              true);
+        if (ret == WT_NOTFOUND && !commit) {
```
Comment (luke-pearson) [inline src/txn/txn.c:936]: This was an else if which we previously couldn't get into but now I think we can, thats fine?

Comment (quchenhao) [reply src/txn/txn.c:936]: The intention is to get into that after we call __txn_append_hs_record

Comment (quchenhao) [reply src/txn/txn.c:936]: We return not found in __txn_append_hs_record if the cursor is placed on the wrong key. In this case, we also need to delete the prepared update.

Comment (luke-pearson) [reply src/txn/txn.c:936]: Thanks makes sense, thanks.

**Pull request merged**